### PR TITLE
Fix pthread_key exhaustion due to mod_spatialite

### PR DESCRIPTION
**Pull request by olsen232**
## Description

In kart, the mod_spatialite extension is loaded once per sqlite session - extensions aren't shared between sessions by default, since subsequent sessions might need different extensions.
This change patches our mod_spatialite so that it doesn't unload itself when the sqlite session ends - it stays loaded (if not necessarily active) until the process exits.
This avoids a leak in pthread_keys that was happening due to repeated loading and unloading of the mod_spatialite extension, which avoids test failures in gdal / raster specific tests caused by gdal crashing once pthread_keys are exhausted.

## Checklist:

- [x] Have you reviewed your own change?
- [x] Have you included test(s)?
- [x] Have you updated the [changelog](https://github.com/koordinates/kart/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ _When adding new entries to the changelog, please include issue/PR numbers where
 
 - Allow kart to import non-spatial tables from PostgreSQL databases which don't have postGIS extension installed. [#728](https://github.com/koordinates/kart/issues/728)
 - Fixes a bug where the current spatial filter isn't stored in the filesystem working copy, affecting the spatial filtering of point cloud datasets. [#833](https://github.com/koordinates/kart/pull/833)
+- Fixes pthread_key leaks in a long running Kart process due to repeated loading and unloading of mod_spatialite. [#838](https://github.com/koordinates/kart/pull/838)
```
Comment (rcoup) [inline CHANGELOG.md:11]: should add the amazonlinux change here too

File: .github/workflows/build.yml
Comment (rcoup) [inline .github/workflows/build.yml:1]: Good change, but this should probably be pulled out of this PR. Maybe cherry-pick the commit to master, then rebase this?

File: tests/raster/fixtures.py [hunk 2]
```diff
@@ -1,28 +1,19 @@
...
-    with data_archive_readonly("raster/cog-aerial.tgz") as aerial:
-        try:
-            gdal.Info(f"{aerial}/aerial.tif")
-            has_gdal_info = True
-        except Exception as e:
-            click.echo(e, err=True)
-            has_gdal_info = False
+        assert gdal.Info is not None
+        has_gdal_info = True
+    except Exception as e:
```
Comment (rcoup) [inline tests/raster/fixtures.py:14]: When would we have `gdal` but not `gdal.Info()`? I know we were trying to track down a bug, but do we need to keep this?

**Review by rcoup — APPROVED** [review]
I think this is good as a workaround in the meantime. Longer-term we should drop mod_spatialite in favour of loading libgdal as a sqlite module for the necessary geopackage support

File: CHANGELOG.md (hunk 1 above)
Comment (olsen232) [reply CHANGELOG.md:11]: Done

File: .github/workflows/build.yml
Comment (olsen232) [reply .github/workflows/build.yml:1]: Done

File: tests/raster/fixtures.py (hunk 2 above)
Comment (olsen232) [reply tests/raster/fixtures.py:14]: Done

**Pull request merged**